src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -42,6 +42,7 @@ public ProjectRestoreCommand(ProjectRestoreRequest request)
 
         public async Task<Tuple<bool, List<RestoreTargetGraph>, RuntimeGraph>> TryRestoreAsync(LibraryRange projectRange,
             IEnumerable<FrameworkRuntimePair> frameworkRuntimePairs,
+            IDictionary<NuGetFramework, string> frameworkToAlias,
```

Comment:
The PR description says:
> The full implementation of aliasing allows multiple TargetFramework with the same effective tfm

But this dictionary can't take two aliases with the same effective NuGetFramework.

Although later in the PR description it says something about "eventually support full aliasing", so the PR description doesn't make it clear to me if this is supposed to implement full aliasing. Now I feel like it's probably just some initial code, so maybe this dictionary is known to need changes before full support is implemented?

Reply:
This is the same response for the next 3 comments as well, the perf nitpicks. It's really the same answer. 

The code path you are commenting on is the legacy resolver. I did end up making the changes to skip the unnecessary allocations, but the reason why I wasn't that worried is because this was the old algorithm that has no real usage.